internal/reconstitution/writebuffer.go

```diff
@@ -129,44 +128,46 @@ func (w *writeBuffer) updateSlice(ctx context.Context, sliceNSN types.Namespaced
 		return false
 	}
 
-	// Allocate the status slice if needed
-	var status *apiv1.ResourceSliceStatus
-	var dirty bool
-	if len(slice.Status.Resources) != len(slice.Spec.Resources) {
-		dirty = true
-		status = slice.Status.DeepCopy()
-		status.Resources = make([]apiv1.ResourceState, len(slice.Spec.Resources))
+	// It's easier to pre-allocate the entire status slice before sending patches
+	// since the "replace" op requires an existing item.
+	if len(slice.Status.Resources) == 0 {
+		copy := slice.DeepCopy()
+		copy.Status.Resources = make([]apiv1.ResourceState, len(slice.Spec.Resources))
+		err = w.client.Status().Update(ctx, copy)
```

Comment:
Why reset the status on every update?

Reply:
This is only reached when the status is uninitialized

Reply:
Ideally the slice wouldn't need to be initialized before applying patches but other controllers assume the resources slice is either nil or len == len(spec.Resources). So it isn't safe to partially initialize the status currently